tests/python/test_multidevice.py

```diff
@@ -112,3 +114,81 @@ def multidevice_schedule(self):
     torch.testing.assert_close(
         out_tensors[0], expected_out_tensor, rtol=1.3e-6, atol=1e-4
     )
+
+
+@pytest.mark.skipif(
+    utils.is_pre_ampere(),
+    reason="Flash Attention is only supported on Ampere and newer devices.",
+)
+@pytest.mark.mpi
+def test_sdpa(mpi_test):
+    d, b, s, a, h = mpi_test.size, 2, 1024, 12, 768
+
+    if a % d != 0:
+        pytest.skip(f"We only support even split, so {a} has to be divisible by {d}.")
+
+    class Model(FusionDefinition):
+        def definition(self) -> None:
+            self.q = self.define_tensor(
+                shape=[d, -1, -1, -1, -1],
+                contiguity=[True if d > 1 else None, True, True, True, True],
+                dtype=DataType.BFloat16,
+            )
+            self.k = self.define_tensor(
+                shape=[d, -1, -1, -1, -1],
+                contiguity=[True if d > 1 else None, True, True, True, True],
+                dtype=DataType.BFloat16,
+            )
+            self.v = self.define_tensor(
+                shape=[d, -1, -1, -1, -1],
+                contiguity=[True if d > 1 else None, True, True, True, True],
+                dtype=DataType.BFloat16,
+            )
+            dropout_p = self.define_scalar(0.0, dtype=DataType.Double)
+            is_causal = self.define_scalar(True, dtype=DataType.Bool)
+            sdpa_result = self.ops.sdpfa_fwd(
+                self.q, self.k, self.v, dropout_p, is_causal, scale=None
+            )
+            attn = sdpa_result[0]
+            self.add_output(attn)
+
+        def multidevice_schedule(self) -> None:
+            mesh = self.sched._create_device_mesh(range(d))
+            for t in [self.q, self.k, self.v]:
+                self.sched._set_device_mesh(t, mesh)
+                self.sched.parallelize(t, 0, nvfuser.ParallelType.mesh_x)
+
+    torch.cuda.set_device(mpi_test.local_rank)
+    torch.manual_seed(0)
```

Comment:
shouldn't we make the seed manual for all tests, and put this line in the test's constructor ?

Reply:
Yes and cc @Priya2698 who briefly mentioned how this should be done which I forgot :) We do have https://github.com/NVIDIA/Fuser/blob/61a77e0a64d5bc446ba1c009f04a19204a28eab2/tests/python/utils.py#L387 a "fixture" class used by many Python tests, so we could set seed there. However, I'm under the impression that we are migrating from unittest to pytest, so this will have to be done differently.

Reply:
We are using this with `pytest`: https://github.com/NVIDIA/Fuser/blob/e01e23ef2f3b4d49e05a5e6a1f9c81119385d537/tests/python/test_python_frontend.py#L41-L42.

What do you mean by doing it differently for pytest?